### PR TITLE
sensors: lps22hb: Enhance driver with trigger

### PR DESCRIPTION
**Pull request by stubb0rnCoder**
Using the `LPS22HH` driver as a basis, this PR replaces the very limited existing driver for the `HB` variant with a more sophisticated one that allows setting a trigger and gives a basis for further improvements.

### CODE REVIEW

File: drivers/sensor/lps22hh/lps22hh.c [hunk 1]
```diff
@@ -181,7 +181,7 @@ static int lps22hh_init_chip(const struct device *dev)
 
 	if (chip_id != LPS22HH_ID) {
 		LOG_ERR("%s: Invalid chip ID 0x%02x", dev->name, chip_id);
-		return -EIO;
+		// return -EIO;
```
Comment (avisconti) [inline drivers/sensor/lps22hh/lps22hh.c:184]: Is this a typo?

**Review by avisconti — APPROVED** [review]
Please rebase. I promise I'll review it. Thanks

**Review by avisconti — CHANGES_REQUESTED** [review]
Also split the work into atomic commits

File: drivers/sensor/lps22hh/lps22hh.c (hunk 1 above)
Comment (stubb0rnCoder) [reply drivers/sensor/lps22hh/lps22hh.c]: yes. I fixed it

**Comment by stubb0rnCoder** [comment]
> Also split the work into atomic commits

Would you like me to split the already commited work or for future work?

PS. This is my first PR to Zephyr and I struggled a bit with signing-off so I ended up with a single commit to fix commits accidently not singed off

File: dts/bindings/sensor/st,lps22hb-common.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      the signal that is presented to the driver.
+
+  odr:
+    type: int
+    default: 0
+    description: |
+      Specify the default output data rate expressed in samples per second (Hz).
+      The default is the power-on reset value.
+
+      - 0  # LPS22HB_DT_ODR_POWER_DOWN
```
Comment (glenn-andrews) [inline dts/bindings/sensor/st,lps22hb-common.yaml:36]: Use samples per second rather than the register value to match the Kconfig

**Review by glenn-andrews — COMMENTED** [review]
I think there are ways to do this that don't break existing code:
1. Either keep the name `st,lps22hb-press` or allow both the new and old names be used.
2. Allow `CONFIG_LPS22HB_SAMPLING_RATE` to be used as the default if `odr` is not defined in the devicetree
3. Remove the I3C option as this is not an I3C device.

**Comment by glenn-andrews** [comment]
Also, if you do make it so that existing code does not need updating, I'd split this into two PRs: One PR for the driver code, which will still work with the existing boards and samples, and a PR updating the boards and samples to the new name.

Multiple smaller PRs are easier to review.

File: dts/bindings/sensor/st,lps22hb-common.yaml (hunk 2 above)
Comment (stubb0rnCoder) [reply dts/bindings/sensor/st,lps22hb-common.yaml]: matched it

File: drivers/sensor/lps22hb/Kconfig [hunk 3]
```diff
@@ -4,16 +4,55 @@
 menuconfig LPS22HB
 	bool "LPS22HB pressure and temperature"
 	default y
-	depends on DT_HAS_ST_LPS22HB_PRESS_ENABLED
-	select I2C
+	depends on DT_HAS_ST_LPS22HB_ENABLED
+	depends on ZEPHYR_HAL_ST_MODULE
+	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ST_LPS22HB),i2c)
+	select I3C if $(dt_compat_on_bus,$(DT_COMPAT_ST_LPS22HB),i3c)
```
[the remark this answers is not in the document]
Comment (stubb0rnCoder) [reply drivers/sensor/lps22hb/Kconfig]: That was an artifact that should not have been committed. I removed it

File: boards/96boards/argonkey/96b_argonkey.dts [hunk 4]
```diff
@@ -150,8 +150,8 @@
 		drdy-gpios = <&gpioa 2 GPIO_ACTIVE_HIGH>;
 	};
 
-	lps22hb-press@5d {
-		compatible = "st,lps22hb-press";
+	lps22hb@5d {
```
[the remark this answers is not in the document]
Comment (stubb0rnCoder) [reply boards/96boards/argonkey/96b_argonkey.dts]: I reverted it to the original naming to change it another day

File: dts/bindings/sensor/st,lps22hb-i3c.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2019 STMicroelectronics
```
[the remark this answers is not in the document]
Comment (stubb0rnCoder) [reply dts/bindings/sensor/st,lps22hb-i3c.yaml]: removed

File: drivers/sensor/lps22hb/Kconfig [hunk 6]
```diff
@@ -4,16 +4,55 @@
...
+	select SPI if $(dt_compat_on_bus,$(DT_COMPAT_ST_LPS22HB),spi)
+	select HAS_STMEMSC
+	select USE_STDC_LPS22HB
 	help
 	  Enable driver for LPS22HB I2C-based pressure and temperature
```
[the remark this answers is not in the document]
Comment (stubb0rnCoder) [reply drivers/sensor/lps22hb/Kconfig]: improved it

File: drivers/sensor/lps22hb/Kconfig [hunk 7]
```diff
@@ -4,16 +4,55 @@
...
 	  sensor.
 
-config LPS22HB_SAMPLING_RATE
```
[the remark this answers is not in the document]
Comment (stubb0rnCoder) [reply drivers/sensor/lps22hb/Kconfig]: I reintroduced it

**Review by glenn-andrews — COMMENTED** [review]
There's a problem with the devicetree bindings. `thingy52_nrf52832.dts` uses the `vin-supply` field included from the `I2C-device.yaml` binding, which is not available in `st,lps22hb-common.yaml`

There's a way of setting it up so devices on the I2C bus take one binding while devices on the SPI bus take the other binding, but I don't have time to look it up right now.

**Comment by glenn-andrews** [comment]
@stubb0rnCoder If you're still around I'd love to work with you on landing this.

**Pull request closed without merging**